lib/oli/delivery/attempts/activity_lifecycle.ex

```diff
@@ -155,7 +155,8 @@ defmodule Oli.Delivery.Attempts.ActivityLifecycle do
                             grading_approach: p.grading_approach,
                             response: response,
                             activity_attempt_id: new_activity_attempt.id,
-                            datashop_session_id: datashop_session_id
+                            datashop_session_id: datashop_session_id,
```

Comment:
The 2 changes in this file carry our hint-state forward on resetting a question.